tt_metal/impl/device/device.cpp

```diff
@@ -1873,13 +1873,15 @@ void Device::init_command_queue_device() {
     this->configure_command_queue_programs();
     Program& command_queue_program = *this->command_queue_programs[0];
 
-    for (uint8_t cq_id = 0; cq_id < this->num_hw_cqs(); cq_id++) {
-        for (const auto &[core_type, logical_dispatch_cores] : command_queue_program.logical_cores()) {
-            for (const CoreCoord &logical_dispatch_core : logical_dispatch_cores) {
-                launch_msg_t msg = command_queue_program.kernels_on_core(logical_dispatch_core, core_type)->launch_msg;
-                CoreCoord phys_core = this->physical_core_from_logical_core(logical_dispatch_core, core_type);
-                tt::llrt::write_launch_msg_to_core(this->id(), phys_core, &msg, this->get_dev_addr(phys_core, HalMemAddrType::LAUNCH));
-            }
+    // TODO: should get a const ref
+    std::vector<std::vector<CoreCoord>>logical_cores = command_queue_program.logical_cores();
+    for (uint32_t index = 0; index < hal.get_programmable_core_type_count(); index++) {
+        const auto& logical_dispatch_cores = logical_cores[index];
```

Comment:
currently we won't ever have some dispatch cores on tensix and others on eth, will we eventually allow something like that?

Reply:
Maybe, but I think unlikley

Reply:
By current construction of dispatch cores we won't fall into that case but this code makes it look like it could be possible

Reply:
yes, this code is generic.  I think that's way to go, make the code generic when possible/not in a perf path